Minikube guide updates

aditighag opened the pull request:
Add a note about `--cni=cilium` flag that can automatically install Cilium. However, it installs v1.8 by default so add a note regarding this.

Cilium init script mounts the BPF fs if not already mounted so the manual step to mount BPF fs isn't required. Let's double check - 

```
$ minikube start --network-plugin=cni
$ cilium install
$ minikube ssh -- mount | grep bpf
none on /sys/fs/bpf type bpf (rw,relatime)

ks logs cilium-zp2st -c ebpf-mount
Mounting eBPF filesystem...
```

`ebpf-mount` is the init container that mounts the BPF fs. Here is the relevant code in cilium/cli - https://github.com/cilium/cilium-cli/blob/master/install/install.go#L685

Fixes: #16324

aanm commented on Documentation/gettingstarted/k8s-install-default.rst:
```diff
@@ -98,7 +98,13 @@ to create a Kubernetes cluster locally or using a managed Kubernetes service:
+
+.. note::
+
+   From minikube v1.12.1+, cilium networking plugin can be enabled directly with
```
Let's bump the minikube version in line 95 to v1.12.1 and remove this note altogether.

aditighag replied:
Yeah, thought about it. Sounds good to me!

aanm commented on Documentation/gettingstarted/k8s-install-default.rst:
```diff
@@ -92,14 +92,22 @@ to create a Kubernetes cluster locally or using a managed Kubernetes service:
+   flag enabled, ``minikube`` will not only mount eBPF file system but also
+   deploy ``quick-install.yaml`` automatically. However, this may not install the
+   latest version of cilium.
+
```
I don't think we should have it, it's misleading and users might think it's an easier path to install Cilium, which it is, but installs an older Cilium version (1.8)

aditighag replied:
I've already added a note about the version getting installed. It's useful to have a simpler step for users who don't care about version, and/or might miss the manual step.
Also, check this note from minikube command - 

>❗  With --network-plugin=cni, you will need to provide your own CNI. See --cni flag as a user-friendly alternative

aditighag commented on Documentation/gettingstarted/k8s-install-default.rst:
```diff
@@ -92,14 +92,22 @@ to create a Kubernetes cluster locally or using a managed Kubernetes service:
 
-          minikube start --network-plugin=cni
+          minikube start --network-plugin=cni --memory=4096
           minikube ssh -- sudo mount bpffs -t bpf /sys/fs/bpf
```
Let's double check if this step is required. Cilium init script mounts the BPF fs if not already mounted. 
I just ran these commands - 

```
$ minikube start --network-plugin=cni
$ cilium install
$ minikube ssh -- mount | grep bpf
none on /sys/fs/bpf type bpf (rw,relatime)


```

nbusseneau replied:
It should not, this is done via `cilium install` now: https://github.com/cilium/cilium-cli/blob/ce6aa0252795e5ec4909f39245bced0cc4cbdf6f/install/install.go#L685

aditighag replied:
That's already mentioned in the commit description. ;)

aanm replied:
I've tested it locally. It works now, I'm not sure why it failed for me before :thinking:

aditighag committed:
Revert "docs: add missing mount bpf fs on minikube GSG"

This reverts commit f3c90b86a51d633741bae712179e609bedfdb7bf.

Cilium init container mounts the BPF fs if it's not already mounted so the 
manual step to mount BPF fs isn't required. For details, check GH #16346.

Signed-off-by: Aditi Ghag <aditi@cilium.io>

aditighag committed:
docs: Add note about --cni=cilium flag

From minikube v1.12.1+, the flag passed to `minikube start` command
will deploy quick-install.yaml automatically. However, it currently installs
v1.8 by default so add a note regarding this.

Signed-off-by: Aditi Ghag <aditi@cilium.io>

qmonnet reviewed (approved):
> Cilium init script mounts the BPF fs if not already mounted so the manual step to mount BPF fs isn't required.

(Note that it could be worth having the above in the commit log of the revert patch. Not blocking, though.)

Looks good, thank you! Don't forget to sign-off :)

aanm commented on Documentation/gettingstarted/k8s-install-default.rst:
```diff
@@ -98,7 +98,12 @@ to create a Kubernetes cluster locally or using a managed Kubernetes service:
+
+   From minikube v1.12.1+, cilium networking plugin can be enabled directly with
+   ``--cni=cilium`` parameter in ``minikube start`` command. However, this may not
+   install the latest version of cilium.
```
I don't think we should have it, it's misleading and users might think it's an easier path to install Cilium, which it is, but installs an older Cilium version (1.8)

aditighag replied:
I've already added a note about the version getting installed. It's useful to have a simpler step for users who don't care about version, and/or might miss the manual step.
Also, check this note from minikube command -

❗ With --network-plugin=cni, you will need to provide your own CNI. See --cni flag as a user-friendly alternative

joestringer replied:
Given this is just a note to basically acknowledge that there is another path (but suggest why we don't advise that path by default), I'm OK with this. However I think it should be hidden inside the minikube group-tab, not at the outermost level of indentation.

aditighag replied:
@joestringer Good catch! Fixed the formatting, PTAL. Thanks!

joestringer reviewed (changes requested):
👍 

I'm ambivalent about documenting the `minikube --cni=cilium ...` flag or not, seems like we have legitimate reasons not to recommend the user to use it so it'd be simpler to omit the instruction. That said it's not a big deal to also just let the user know like this. However I think if we do so, we should move the note inside the group tab for minikube users rather than instruct all users about this, even if they've clicked one of the other group tabs (why would cloud or kind users need to be told about a minikube feature?).
